models/question.js

```diff
@@ -0,0 +1,49 @@
+"use strict";
+
+const db = require("../db");
+const ExpressError = require("../expressError");
+
+/** Methods for questions. */
+
+class Question {
+  /** Get questions by subjectId. */
+
+  static async getBySubject(subjectId) {
+    // check for subject specific questions first
+    const result = await db.query(
+      `SELECT q.id, q.text FROM questions AS q
+        JOIN questions_subjects AS qs ON q.id = qs.question_id
+        WHERE qs.subject_id = $1`,
+      [subjectId]
+    );
+    const questions = result.rows;
+
+    // check for school questions if no subject questions are found
+    if (questions.length === 0) {
```

Comment:
true, I should have just used `if (!questions.length)`

Reply:
yup, if you wanted to increase readability even more, I like using loDash so I could do something like:
https://lodash.com/docs/#isEmpty
```
if (isEmpty(questions))
```

it's sometimes easier for us to reason on a positive evaluation rather than a negative one.